### DISCUSSION TITLE
v-map component

### DISCUSSION THREAD

**Discussion by michaeltoohig** [directus/directus#22906] [Feature Requests]
### Summary

I request a `v-map` component so that the Directus map interface can be used in custom extensions to provide a consistent experience.

I see there is the interface map component and the layout map component in the Directus codebase already but I am looking for a means to import a map component into my module so that I can allow my users to select a LngLat/Coordinates without having rebuild the whole map component myself and install `maplibre-gl` in my own custom extension.

### Basic Example

```
<v-map
  v-model="data"
  :type="'geometry'"
  :geometry-type="geometry.Point"
  :loading="loading"
  :disabled="disabled"
/>
```

### Motivation

Prevents myself and perhaps other extension authors from re-inventing the wheel. Provides a consistent Directus experience in extensions which use coordinates or other GeoJSON like data.

### Detailed Design

I'm just looking for something that is relatively generic but sets up the map and style and transitions and all that polish provided in the existing Directus map interface.

```
<v-map
  v-model="data"
  :type="'geometry'"
  :geometry-type="geometry.Point"
  :loading="loading"
  :disabled="disabled"
/>
```

### Must Have:
- All the usual props for inputs `disabled`, `loading`, etc.
- Use the existing map style dropdown found in the map interface to change the map style using the global map style settings

### Should Have:
- prop for the `v-model` or atleast an `emit` for marker position or some other polygon positioning change event.

### Could Have:
- props to set the initial center and zoom levels perhaps these values are bound
- options to disable different map features such as zoom or GPS locate, etc.

### Drawbacks

It may add burden for Directus team to implement given they already have 2 map components in the codebase; however, it would be no different than existing Directus interfaces that use `v-` components under the hood already like the `select-dropdown` interface uses the `v-select` component.

### Alternatives

Building custom map components for every Directus extension author who wants a map interface for user input.

### Adoption Strategy

It will become a newly available component `v-map` or something like that.

**Comment by michaeltoohig** [comment]
So further searching I did find this https://github.com/directus/directus/discussions/12377#discussioncomment-2938491 however I have not yet had time to really test it. I now have the interface in my custom extension but its not rendering and I run into issues with the props still as I haven't figured out all the required props yet. No matter I'm not sure this is a supported pattern or well documented so I wanted to bring attention to it.

**Reply by hanneskuettner** [reply]
You should be able to just use `<interface-map />` in your extension (without having to import them) and provide it with the props you can find in it's implementation https://github.com/directus/directus/blob/main/app/src/interfaces/map/map.vue 

Generally all interfaces/displays/layouts are available to be used in extensions as they are globally registered on the Vue app.

**Reply by michaeltoohig** [reply]
Okay, so that's good to know but I had no idea how I would learn about that. Would it be possible to include these in the [components documentation](https://components.directus.io) for reference and discovery? But, I'm guessing that this components documentation website requires manual work and is not automatically generated because it doesn't include all of the `v-` components such as `v-drawer` or `v-table` and it includes some funny examples such as [this](https://components.directus.io/story/src-components-v-sheet-story-vue?variantId=_default). So including additional components such as the `interface-` components is probably asking a lot.

But now I have to go experiment with the `interface-` components as I may be able to use them more in my module rather than recreating my own `v-input` wrapper component and others.
